reVX/hybrid_stats/hybrid_stats.py

```diff
@@ -0,0 +1,829 @@
+            columns_map[s] = cols
+            columns.extend(list(cols.values()))
+
+        return columns_map, columns
+
+    @classmethod
+    def _compute_stats(cls, solar_data, wind_data, statistics,
+                       diurnal=False, month=False):
+        """
+        Compute desired stats for desired time intervals from res_data
+
+        Parameters
+        ----------
+        solar_data : pandas.DataFrame
+            DataFrame of solar data. Index is time_index, columns are sites
+        wind_data : pandas.DataFrame
+            DataFrame of wind data. Index is time_index, columns are sites
+        statistics : dict
+            Dictionary of statistic functions/kwargs to run
+        diurnal : bool, optional
+            Extract diurnal stats, by default False
+        month : bool, optional
+            Extract monthly stats, by default False
+
+        Returns
+        -------
+        out_stats : pandas.DataFrame
+            DataFrame of desired statistics at desired time intervals
+        """
+        sites = solar_data.columns.values
+        solar_data = cls._groupby_data(solar_data, diurnal=diurnal,
+                                       month=month)
+        wind_data = cls._groupby_data(wind_data, diurnal=diurnal,
+                                      month=month)
+
+        cols_map, col_names = cls._create_names(list(solar_data.groups),
+                                                list(statistics))
+        out_stats = pd.DataFrame(columns=col_names, index=sites,
+                                 dtype=np.float32)
+        for grp_name, solar_grp in solar_data:
+
+            wind_grp = wind_data.get_group(grp_name)
+            msg = ('solar and wind data shapes do not match! {} != {}'
+                   .format(solar_grp.shape, wind_grp.shape))
+            assert solar_grp.shape == wind_grp.shape, msg
+            for i in sites:
+                solar_ts = solar_grp.iloc[:, i]
+                wind_ts = wind_grp.iloc[:, i]
+                for name, stat in statistics.items():
+                    col = cols_map[name][grp_name]
+                    func = stat['func']
+                    kwargs = stat.get('kwargs', {})
+                    out_stats.at[i, col] = func(solar_ts, wind_ts, **kwargs)
+
+        return out_stats
+
+    @classmethod
+    def _extract_stats(cls, solar_h5, wind_h5, sites, statistics, dataset,
+                       time_index, res_cls=Resource, hsds=False,
+                       diurnal=False, month=False, combinations=False):
+        """
+        Extract stats for given dataset, sites, and temporal extent
+
+        Parameters
+        ----------
+        solar_h5 : str
+            Path to solar h5 file(s)
+        wind_h5 : str
+            Path to wind h5 file(s)
+        sites : pandas.DataFrame
+            Subset of meta DataFrame with sites to extract
+        statistics : dict
+            Dictionary of statistic functions/kwargs to run
+        dataset : str
```

Comment:
Would you want to open this up to a solar_dataset and wind_dataset so you could do it on the Resource data if necessary?